packages/alignments/src/PileupRenderer/PileupRenderer.ts

```diff
@@ -37,17 +46,79 @@ interface LayoutRecord {
   heightPx: number
 }
 
+interface RenderArgsSoftClip extends RenderArgsDeserialized {
+  showSoftClip?: boolean
+}
+
+interface PileupLayoutSessionProps extends LayoutSessionProps {
+  sortObject: unknown
+  showSoftClip: unknown
+}
+
+interface CachedPileupLayout extends CachedLayout {
+  sortObject: unknown
+  showSoftClip: unknown
+}
+
+class SoftClipLayoutSession extends LayoutSession {
+  sortObject: unknown
+
+  showSoftClip: unknown
+
+  cachedLayoutIsValid(cachedLayout: CachedPileupLayout) {
```

Comment:
This should probably use super.cachedLayoutIsValid(cachedLayout) + the logic specific to this